swan-lake/resources/featured-scenarios/write-a-restful-api-with-ballerina.md

```diff
@@ -157,36 +274,30 @@ resource function post countries(@http:Payload CovidEntry[] covidEntries)
 
 In this code:
 
-- It is chosen to either accept the entire payload or send back an error. Copying this straightway results in an error, which is expected as the `ConflictingIsoCodesError` type is not defined yet.
-- This resource has a resource argument named `covidEntries` annotated with `@http:Payload`. This means the resource is expecting a payload with the `CovideEntry[]` type. There are two types of records `CovideEntry[]` and `ConflictingIsoCodesError` that will be used as the return values.
+- It is chosen to accept the entire payload or send back an error. Copying this straightway results in an error, which is expected as the `ConflictingIsoCodesError` type is not defined yet.
+- This resource has an argument named `covidEntries` annotated with `@http:Payload`, which means the resource expects a payload with the `CovideEntry[]` type. Two types of records, `CovideEntry[]` and `ConflictingIsoCodesError`, will be used as the return values.
+
+## Implement the second endpoint
 
-#### Define the error records
+The second endpoint has only one resource to get COVID-19 data filtered by the ISO code.
 
-To define the error records, add the code below to the API template file (i.e., `service.bal`).
+### Define the error detail record
+
+Similar to how you created the error detail record in [define the error detail records](#define-the-error-detail-records), define the error detail record below of the second endpoint using the diagram view in VS Code.
 
 ```ballerina
-public type ConflictingIsoCodesError record {|
-    *http:Conflict;
+public type InvalidIsoCodeError record {|
+    *http:NotFound;
     ErrorMsg body;
 |};
-
-public type ErrorMsg record {|
-    string errmsg;
-|};
 ```
 
 In this code:
-- Ballerina uses `*http:Conflict` to denote that one type is a subtype of another. In this case, `ConflictingIsoCodesError` is a subtype of `http:Conflict`.
-- The body of the response is of type `ErrorMsg`, which simply has a string field named `errmsg`. Based on the need, users can have any data type for their response body.
-- Ballerina has a defined set of types for each HTTP status code. This allows you to write services in a type-oriented way, which in turn is helpful when it comes to tooling and generating OpenAPI specifications for HTTP services. 
-
-## Implement the second endpoint
-
-The second endpoint has only one resource to get COVID-19 data filtered by the ISO code.
+- As in the previous example, this resource also includes its return types. However, the basic principle behind them is as the previous example. 
```

Comment:
" resource also includes its return" is not clear IMO. Do we need this at all?

Reply:
Removed.